repo/blob/s3/s3_storage_test.go

```diff
@@ -61,40 +103,92 @@ func endpointReachable() bool {
 	return false
 }
 
-func TestS3Storage(t *testing.T) {
-	testStorage(t, accessKeyID, secretAccessKey, "")
+func TestS3StorageAWS(t *testing.T) {
+	t.Parallel()
+	// skip the test if AWS creds are not provided
+	options := &Options{
+		Endpoint:        getEnv(testEndpointEnv, awsEndpoint),
+		AccessKeyID:     getEnvOrSkip(t, testAccessKeyIDEnv),
+		SecretAccessKey: getEnvOrSkip(t, testSecretAccessKeyEnv),
+		BucketName:      getEnvOrSkip(t, testBucketEnv),
+		Region:          getEnvOrSkip(t, testRegionEnv),
+	}
+
+	testStorage(t, options)
 }
 
-func TestS3StorageWithSessionToken(t *testing.T) {
-	// create kopia user and session token
-	createUser(t)
-	kopiaAccessKeyID, kopiaSecretKey, kopiaSessionToken := createTemporaryCreds(t)
-	testStorage(t, kopiaAccessKeyID, kopiaSecretKey, kopiaSessionToken)
+func TestS3StorageAWSSTS(t *testing.T) {
+	t.Parallel()
+	// skip the test if AWS creds are not provided
+	options := &Options{
+		Endpoint:        getEnv(testEndpointEnv, awsEndpoint),
+		AccessKeyID:     getEnvOrSkip(t, testSTSAccessKeyIDEnv),
+		SecretAccessKey: getEnvOrSkip(t, testSTSSecretAccessKeyEnv),
+		SessionToken:    getEnvOrSkip(t, testSessionTokenEnv),
+		BucketName:      getEnvOrSkip(t, testBucketEnv),
+		Region:          getEnvOrSkip(t, testRegionEnv),
+	}
+
+	testStorage(t, options)
+}
+
+func TestS3StorageMinio(t *testing.T) {
+	t.Parallel()
+
+	options := &Options{
+		Endpoint:        minioEndpoint,
+		AccessKeyID:     minioAccessKeyID,
+		SecretAccessKey: minioSecretAccessKey,
+		BucketName:      minioBucketName,
+		Region:          minioRegion,
+	}
+
+	if !endpointReachable(options.Endpoint) {
+		t.Skip("endpoint not reachable")
+	}
+
+	createMinioBucket(t, options.BucketName)
+	testStorage(t, options)
 }
 
-func testStorage(t *testing.T, accessID, secretKey, sessionToken string) {
-	if !endpointReachable() {
+func TestS3StorageMinioSTS(t *testing.T) {
+	t.Parallel()
+	// create kopia user and session token
+	kopiaUserName := generateName("kopiauser")
+	kopiaUserPasswd := generateName("kopiapassword")
+
+	createMinioUser(t, kopiaUserName, kopiaUserPasswd)
+	kopiaAccessKeyID, kopiaSecretKey, kopiaSessionToken := createMinioSessionToken(t, kopiaUserName, kopiaUserPasswd, minioBucketName)
+
+	options := &Options{
+		Endpoint:        minioEndpoint,
+		AccessKeyID:     kopiaAccessKeyID,
+		SecretAccessKey: kopiaSecretKey,
+		SessionToken:    kopiaSessionToken,
+		BucketName:      minioBucketName,
+		Region:          minioRegion,
+	}
+
+	if !endpointReachable(options.Endpoint) {
 		t.Skip("endpoint not reachable")
 	}
 
-	ctx := context.Background()
+	createMinioBucket(t, options.BucketName)
```

Comment:
Similarly, it makes sense (to attempt) to create the bucket in S3 tests as well. If an already existing bucket is specified, this should be a no-op. Well, the implementation below will get an error and ignore it, so it should continue. Essentially, it is desirable to run the test without having to create the bucket and have the test create it, and also be able to reuse the bucket. There is no need (and likely not desirable) to remove the bucket afterwards.